src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -315,19 +316,21 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
   }
 
   def updateWorkspace(workspaceName: WorkspaceName, operations: Seq[AttributeUpdateOperation]): Future[PerRequestMessage] =
-    dataSource.inTransaction { dataAccess =>
-      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Write, dataAccess) { workspaceContext =>
-        val updateAction = Try {
-          val updatedWorkspace = applyOperationsToWorkspace(workspaceContext.workspace, operations)
-          dataAccess.workspaceQuery.save(updatedWorkspace)
-        } match {
-          case Success(result) => result
-          case Failure(e: AttributeUpdateOperationException) =>
-            DBIO.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.BadRequest, s"Unable to update ${workspaceName}", ErrorReport(e))))
-          case Failure(regrets) => DBIO.failed(regrets)
-        }
-        updateAction.map { savedWorkspace =>
-          RequestComplete(StatusCodes.OK, savedWorkspace)
+    tryIsCurator(userInfo.userEmail) flatMap { isLibraryCurator =>
+      dataSource.inTransaction { dataAccess =>
+        withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Write, dataAccess) { workspaceContext =>
+          val updateAction = Try {
+            val updatedWorkspace = applyOperationsToWorkspace(workspaceContext.workspace, operations, isLibraryCurator)
```

Comment:
`applyOperationsToWorkspace` also now validates attribute namespace.  This catches deletion of library attributes as well as addition, which is also caught by `workspaceQuery.save`